app/ai/itinerary_nodes/hotels.py

```
@@ -0,0 +1,33 @@
+
+def node_search_hotels(state: PlanState) -> PlanState:
+    dests = state.get("destinations") or []
+    dr = state.get("date_range")
+    if not dests or not isinstance(dr, dict) or not dr.get("start") or not dr.get("end"):
+        state["hotel_options"] = []
+        state["next"] = "finalize"
+        return state
+    # Build typed request; ensure valid dates
+    try:
+        dates = DateRange(**dr)  # type: ignore
+    except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)